src/plugin.py

```diff
@@ -1,113 +1,264 @@
-import sys
+import sys, asyncio, logging, urllib, os, json, pickle, webbrowser
+from galaxy.api.plugin import (
+    Plugin,
+    LocalGame,
+    Authentication,
+    Game,
+    create_and_run_plugin,
+    GameTime,
+)
+from galaxy.api.consts import LocalGameState, LicenseType, OSCompatibility, Platform
+from galaxy.api.types import LicenseInfo
+from galaxyutils import time_tracker
 
-from galaxy.api.plugin import Plugin, create_and_run_plugin
-from galaxy.api.consts import Platform
-from galaxy.api.types import Authentication, Game, LicenseInfo, LicenseType, LocalGame, LocalGameState
+from local import WindowsLocalClient, MacLocalClient
+from consts import (
+    GameID,
+    MINECRAFT_DUNGEONS_INSTALL_URL,
+    MINECRAFT_MAC_INSTALL_URL,
+    MINECRAFT_WIN_INSTALL_URL,
+    GAME_NAMES,
+    IS_WINDOWS,
+    INSTALLED_FOLDER_PATH,
+)
+import utils
+import multimc
+from decorators import double_click_effect
 from version import __version__
 
-import logging as log
-import os
-import subprocess
-from local import LocalClient
-import asyncio
-import requests
-import tempfile
-import shutil
-from typing import Optional, Any, List
+
+log = logging.getLogger(__name__)
 
 
 class MinecraftPlugin(Plugin):
     def __init__(self, reader, writer, token):
         super().__init__(Platform.Minecraft, __version__, reader, writer, token)
-        self.local_client = LocalClient()
-        self.minecraft_launcher = None
-        self.minecraft_uninstall_command = None
-        self.minecraft_installation_status = LocalGameState.None_
-        self.minecraft_running_check = None
-        self.tick_count = 0
+        self.local_client = WindowsLocalClient() if IS_WINDOWS else MacLocalClient()
+        self.status = {
+            GameID.Minecraft: LocalGameState.None_,
+            GameID.MinecraftDungeons: LocalGameState.None_,
+        }
+        self.was_game_launched_task: asyncio.Task = None
+        self.update_task: asyncio.Task = None
+        self.check_sizes_task: asyncio.Task = None
+        self.owned = []
+        self.multimc: multimc.MultiMCClient = None
 
     async def authenticate(self, stored_credentials=None):

```

Comment:
I'm not sure how to use that but I'll definitely have a look at that.

Reply:
Just found this: https://github.com/summer/mojang
I could use this to auto select Minecraft and show users name in the integrations window in settings. There doesn't appear to be a way to see if Minecraft dungeons is installed

Reply:
> You probably meant "is owned", not "is installed"?

Yep. Sorry about that :) 

> wow, this you just found it and it was created yesterday

That's odd. It definitely wasn't created yesterday but for some reason, it says the repo was created yesterday with one commit? Very very weird. It still has the same full documentation and from a quick glance it looks like nothing has changed since I found it *more than a week ago* :). 

> If the web page knows if you have bought a game there is way to figure out ownership.

Yep, just gotta figure out how.